ydb/library/yql/udfs/common/knn/knn.cpp

```diff
@@ -0,0 +1,248 @@
+#include <ydb/library/yql/public/udf/udf_helpers.h>
+
+#include <util/generic/buffer.h>
+#include <util/stream/format.h>
+
+using namespace NYql;
+using namespace NYql::NUdf;
+
+void EnumerateVector(const TUnboxedValuePod vector, std::function<void(float)> callback) {
+    const auto elements = vector.GetElements();
+    if (elements) {
+        const auto size = vector.GetListLength();
+        
+        for (ui32 i = 0; i < size; ++i) {
+            callback(elements[i].Get<float>());
```

Comment:
I'm afraid, it can't be optimized out, so you'll get.real function call for each element. That ruines performace

Reply:
7e85069ac2adee919e1f7f7f7c952d5fdb1061e1